internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -274,6 +274,14 @@ func (r *Reconciler) reconcileControlPlane(ctx context.Context, cluster *cluster
 		}
 	}
 
```

Comment:
One noteworthy diff to the InfraCluster implementation is that we only do this after the infracluster is ready
```go
	if !ready {
		log.V(3).Info("Infrastructure provider is not ready yet")
		return ctrl.Result{}, nil
	}
```

Should we do the same here?